### ISSUE TITLE
Custom vendor-dir and install paths are not recognised.

### ISSUE THREAD

**Issue by kitsunet** [maglnet/ComposerRequireChecker#81]
Doesn't even need to be automatic, if that became configurable instead of using the harcoded `vendor` here https://github.com/maglnet/ComposerRequireChecker/blob/master/src/ComposerRequireChecker/FileLocator/LocateComposerPackageDirectDependenciesSourceFiles.php#L15

that would be super helpful :)

If that's something you can agree with I could look into creating a PR.

**Comment by Ocramius** [comment]
Overall understandable request, but really, I'd rather throw an exception that says it is unsupported. I even throw exceptions when I find a custom `"vendor-dir"` (on other projects of mine). It breaks so many assumptions that you'd really need to go hunting around the ecosystem in general.

**Comment by kitsunet** [comment]
Yep, I can see that. I am fine with that. Took me a bit to figure out why it literally didn't find any of the dependencies in a project were that is all custom...
I guess having a "not supported" message would be something :)

Maybe I create just a super custom version for that.

**Comment by Ocramius** [comment]
> Took me a bit to figure out why it literally didn't find any of the dependencies in a project were that is all custom...

Instead of detecting from `composer.json`, we may ask the user for a path - probably better? Would allow testing projects with multiple composer installations too.

**Comment by kitsunet** [comment]
Sure, that would work!

**Comment by SvenRtbg** [comment]
Supporting `vendor-dir` should be easy to do. The mentioned code already gets the `composer.json` path to the main project and already reads that file to scan for `"require"` dependencies.

Adding to read the value of `["config"]["vendor-dir"]` and assuming `"vendor"` as default shouldn't be too much hassle. Note that this config value is "root-only", so it basically is just using one fixed or one dynamic directory name.

Haven't looked into `install-dir`, but because this is per-package required, I'd rather just emit a notice if possible.

What do you think? Adding support for `vendor-dir`?

**Comment by SvenRtbg** [comment]
Here we are, I have a patch proposal.

**Comment by SvenRtbg** [comment]
While I'm at it: Why does this project allow `minimum-stability:dev` for it's dependencies? I get it from the commit history that once the nikic/php-parser version 2 required was unstable, but this never got reverted even though stable versions are available for eveything.

**Comment by Ocramius** [comment]
@SvenRtbg separate patch if you want - I think it's just a leftover :)
